fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/EliminateAggRule.java

```diff
@@ -0,0 +1,153 @@
+                .addChildren(Pattern.create(OperatorType.PATTERN_LEAF)));
+    }
+
+    public static EliminateAggRule getInstance() {
+        return INSTANCE;
+    }
+
+    private static final EliminateAggRule INSTANCE = new EliminateAggRule();
+
+    @Override
+    public boolean check(OptExpression input, OptimizerContext context) {
+        LogicalAggregationOperator aggOp = input.getOp().cast();
+        if (aggOp.getGroupingKeys().isEmpty()) {
+            return false;
+        }
+
+        for (Map.Entry<ColumnRefOperator, CallOperator> entry : aggOp.getAggregations().entrySet()) {
+            String fnName = entry.getValue().getFnName();
+            if (!(fnName.equals(FunctionSet.SUM) || fnName.equals(FunctionSet.COUNT) ||
+                    fnName.equals(FunctionSet.AVG) ||
+                    fnName.equals(FunctionSet.FIRST_VALUE) ||
+                    fnName.equals(FunctionSet.MAX) || fnName.equals(FunctionSet.MIN) ||
+                    fnName.equals(FunctionSet.GROUP_CONCAT))) {
+                return false;
+            }
+        }
+
+        // collect uk pk key
+        UKFKConstraintsCollector collector = new UKFKConstraintsCollector();
+        input.getOp().accept(collector, input, null);
+
+        OptExpression childOptExpression = input.inputAt(0);
+        for (ColumnRefOperator columnRefOperator : aggOp.getGroupingKeys()) {
+            if (childOptExpression.getConstraints().getUniqueConstraint(columnRefOperator.getId()) == null) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    @Override
+    public List<OptExpression> transform(OptExpression input, OptimizerContext context) {
+        LogicalAggregationOperator aggOp = input.getOp().cast();
+
+        Map<ColumnRefOperator, ScalarOperator> newProjectMap = new HashMap<>();
+
+        for (Map.Entry<ColumnRefOperator, CallOperator> entry : aggOp.getAggregations().entrySet()) {
+            String fnName = entry.getValue().getFnName();
+            ScalarOperator newOperator = handleAggregationFunction(fnName, entry.getValue());
+            if (newOperator != null) {
+                newProjectMap.put(entry.getKey(), newOperator);
+            }
+        }
+
+        LogicalProjectOperator newProjectOp = LogicalProjectOperator.builder().setColumnRefMap(newProjectMap).build();
+        return List.of(OptExpression.create(newProjectOp, input.inputAt(0).getInputs()));
+    }
+
+    private ScalarOperator handleAggregationFunction(String fnName, CallOperator callOperator) {
+        if (fnName.equals(FunctionSet.COUNT)) {
+            return rewriteCountFunction(callOperator);
+        } else if (fnName.equals(FunctionSet.SUM) || fnName.equals(FunctionSet.AVG) ||
+                fnName.equals(FunctionSet.FIRST_VALUE) || fnName.equals(FunctionSet.MAX) ||
+                fnName.equals(FunctionSet.MIN) || fnName.equals(FunctionSet.GROUP_CONCAT)) {
```

Comment:
the return type of sum/avg may different from its arg type. need add cast if necessary.